### PR TITLE
Three Solver-related Stories.

### PR DESCRIPTION
**Pull request by Kenneth-T-Moore**
1. Removed the return values (fail, abs rel) from the entire openmdao solve/solve_*/run_* hierarchy; AnalysisError should be used to detect and act on errors.
2. Replaced vector deepcopy in the debug printing with a lightweight copy that only deepcopies the views.
3. Added an error message for DirectSolver with csc assembled jacobian for when there a row or column in the matrix is a linear  combination of other rows/cols.

### CODE REVIEW

File: openmdao/utils/general_utils.py [hunk 1]
```diff
@@ -506,14 +506,16 @@ def pad_name(name, pad_num=10, quotes=False):
             return '{0}'.format(name)
 
 
-def run_model(prob):
+def run_model(prob, allow_exception=False):
```
Comment (naylor-b) [inline openmdao/utils/general_utils.py]: would 'ignore_exception' be clearer here?

File: openmdao/vectors/vector.py [hunk 2]
```diff
@@ -213,6 +214,22 @@ def _clone(self, initialize_views=False):
...
+    def _copy_views(self):
+        """
+        Return a lightweight copy containing just the views.
+
+        Returns
+        -------
+        <Vector>
+            Vector containing _name, _typ, and _views.
+        """
+        vec_cp = self.__class__(self._name, self._kind, self._system, self._root_vector,
```
Comment (naylor-b) [inline openmdao/vectors/vector.py]: do we really need to recreate the whole Vector here?  We just need to copy the views of the existing vector, right?  Calling __init__ on the Vector calls _initialize_data() and _initialize_views() which do quite a bit of stuff that seems unnecessary here.

File: openmdao/utils/general_utils.py (hunk 1 above)
Comment (Kenneth-T-Moore) [reply openmdao/utils/general_utils.py]: Changed to `ignore_exception`

File: openmdao/vectors/vector.py (hunk 2 above)
Comment (Kenneth-T-Moore) [reply openmdao/vectors/vector.py]: Made a few adjustments so that we just need to save a copy of the views dictionary, then made the suggested change.

**Pull request merged**